### PR TITLE
replaced fetch to rest manager

### PR DESCRIPTION
**Pull request by festinalente91**
- I couldn't test it on the real device, so until then please just checkout the code.

### CODE REVIEW

File: shared/utils.js [hunk 1]
```diff
@@ -43,27 +48,39 @@ export const HTTPUtil = {
     return fetch(address, {
       method: 'GET',
       headers
-    }).then(response => response.json());
+    }).then(response => {
+      if(!response.ok) throw Error(response.json().error);
+      return response.json()
```
Comment (ochanje210) [inline shared/utils.js]: don't you need semicolon at the end of line here?

File: shared/utils.js [hunk 2]
```diff
@@ -43,27 +48,39 @@ export const HTTPUtil = {
...
   },
   post: (address, headers, body) => {
     return fetch(address, {
       method: 'POST',
       headers,
       body: JSON.stringify(body)
-    }).then(response => response.json());
+    }).then(response => {
+      if(!response.ok) throw Error(response.json().error);
+      return response.json()
```
Comment (ochanje210) [inline shared/utils.js]: ditto

File: shared/utils.js [hunk 3]
```diff
@@ -43,27 +48,39 @@ export const HTTPUtil = {
...
   },
   put: (address, headers, body) => {
     return fetch(address, {
       method: 'PUT',
       headers,
       body: JSON.stringify(body)
-    }).then(response => response.json());
+    }).then(response => {
+      if(!response.ok) throw Error(response.json().error);
+      return response.json()
```
Comment (ochanje210) [inline shared/utils.js]: ditto

File: shared/utils.js [hunk 4]
```diff
@@ -43,27 +48,39 @@ export const HTTPUtil = {
...
   },
   delete: (address, headers, body) => {
     return fetch(address, {
       method: 'DELETE',
       headers,
       body: JSON.stringify(body)
-    }).then(response => response.json());
+    }).then(response => {
+      if(!response.ok) throw Error(response.json().error);
+      return response.json()
```
Comment (ochanje210) [inline shared/utils.js]: ditto

File: shared/actions/authActions.js [hunk 5]
```diff
@@ -109,50 +59,25 @@ export const grantAnonymousUser = async (secret, userKey) => {
       }
     })
     .catch((error) => {
+      if (error.message === 'wrong secret') {
```
Comment (ochanje210) [inline shared/actions/authActions.js]: ditto

File: shared/actions/authActions.js [hunk 6]
```diff
@@ -109,50 +59,25 @@ export const grantAnonymousUser = async (secret, userKey) => {
...
-      if (rjson) {
-        setAccessToken(rjson.accessToken);
-        setRefreshToken(rjson.refreshToken);
-        setUserKey(rjson.userKey);
-      } else {
-        return signupFacebookUser(facebookToken);
-      }
+      setAccessToken(rjson.accessToken);
+      setRefreshToken(rjson.refreshToken);
+      setUserKey(rjson.userKey);
```
Comment (ochanje210) [inline shared/actions/authActions.js]: what if the condition of (rjson) is false but it does not throw any error and pass it to catch? then signupFacebookUser would be not called

File: shared/actions/authActions.js [hunk 7]
```diff
@@ -231,94 +156,46 @@ export const setProfileImgUrl = (profileImgUrl) => {
...
-  .then((rjson) => {
-    setAccessToken(rjson.accessToken);
-    setRefreshToken(rjson.refreshToken);
-  })
-  .catch((error) => {
-    console.log(error);
-  })
-};
-
-export const getUserInformation = async (userKey, accessToken) => {
```
Comment (ochanje210) [inline shared/actions/authActions.js]: where did this go?

File: shared/utils.js [hunk 8]
```diff
@@ -18,7 +20,10 @@ export const DEFAULT_HEADERS = {
   'Content-Type': 'application/json'
 };
 
-export const getAuthHeaders = (accessToken) => {
+export const getAuthHeaders = async (accessToken) => {
+  if (!accessToken) {
+    accessToken = await AsyncStorage.getItem(`${STORAGE_NAME}${STORAGE_KEY.ACCESS_TOKEN}`);
+  }
   const headers = JSON.parse(JSON.stringify(DEFAULT_HEADERS));
   headers.Authorization =  `bearer ${accessToken}`;
```
Comment (ochanje210) [inline shared/utils.js]: double space between '=' and ```

File: shared/actions/authActions.js (hunk 5 above)
Comment (ovekyc) [inline shared/actions/authActions.js]: ditto

File: shared/actions/authActions.js [hunk 9]
```diff
@@ -109,50 +59,25 @@ export const grantAnonymousUser = async (secret, userKey) => {
...
+        signupGuestUser();
+      }
       console.log(error);
     });
-  } catch (error) {
-    console.log(error);
-  }
 };
 
+// @TODO need to change this function's name.
```
Comment (ovekyc) [inline shared/actions/authActions.js]: @festinalente91

File: shared/actions/authActions.js [hunk 10]
```diff
@@ -26,81 +17,40 @@ const STORAGE_KEY_userKey = 'userKey';
...
+export const signupFacebookUser = (FacebookToken) => {
+  return UserRESTManager.signupFacebook(FacebookToken)
+    .then((rjson) => {
+      setAccessToken(rjson.accessToken);
+      setRefreshToken(rjson.refreshToken);
+      setUserKey(rjson.userKey);
+      return null;
+    })
+    .catch((error) => {
+      if(error.message === 'Already exist.') {
```
Comment (ochanje210) [inline shared/actions/authActions.js]: thanks

File: shared/utils.js (hunk 1 above)
Comment (festinalente91) [inline shared/utils.js]: right

File: shared/actions/authActions.js (hunk 10 above)
Comment (festinalente91) [inline shared/actions/authActions.js]: right. We need to talk about this.

File: shared/actions/authActions.js (hunk 9 above)
Comment (ochanje210) [inline shared/actions/authActions.js]: agreed

**Pull request merged**